### PR TITLE
Detect security groups based on regex

### PR DESCRIPTION
**Pull request by feikesteenbergen**
Recently, Odd changed it's security group, and also the Group Name of its security group.
We cannot now rely on fixed values anymore, therefore we use a regex to find the group.

It fixes the current behaviour, however it is probably not the optimal solution

### CODE REVIEW

File: acid/senza/templates/base.py [hunk 1]
```diff
@@ -19,7 +20,8 @@
 POSTGRES_PORT = 5432
 HEALTHCHECK_PORT = 8008
 SPILO_IMAGE_ADDRESS = "registry.opensource.zalan.do/acid/spilo-9.5"
-ODD_SG_NAME = 'Odd (SSH Bastion Host)'
+ODD_SG_GROUP_NAME_REGEX = 'Odd.*'
```
Comment (avaczi) [inline acid/senza/templates/base.py]: and then let's hope someone doesn't have the brilliant idea of naming an instance like this ;)

File: acid/senza/templates/base.py [hunk 2]
```diff
@@ -639,19 +638,14 @@ def detect_eu_team_odd_instances(team_zone_name):
...
 
-    if len(zmon_sgs) > 1:
-        fatal_error("More than one security group found for zmon")
+    if len(sgs) == 0:
+        fatal_error('Could not find security group which matches regex {}'.format(sg_regex))
+    if len(sgs) > 1:
+        fatal_error('More than one security group found for regex {}'.format(sg_regex))
 
-    return zmon_sgs[0]
+    return sgs[0]['GroupId']
```
Comment (avaczi) [inline acid/senza/templates/base.py]: This doesn't really handle the issue with multiple Odd hosts (or zmon appliances).  Is there a plan to handle that?

Comment (feikesteenbergen) [inline acid/senza/templates/base.py]: It does, as the lines before raise fatal errors of the lenght != 1

**Comment by avaczi** [comment]
:+1:

**Comment by feikesteenbergen** [comment]
👍

**Pull request merged**